files/templates/additional-scrape-configs.template.yaml

```diff
@@ -180,3 +180,30 @@
   - target_label: provider
     replacement: {{ .Provider }}
 {{ end }}
+# Add kubelet configuration
+- job_name: {{ .ClusterID }}-kubelet
+  honor_timestamps: false
+  scheme: https
+  kubernetes_sd_configs:
+  - api_server: https://{{ .APIServerURL }}
+    role: node
+    tls_config:
+      ca_file: /etc/prometheus/secrets/{{ .SecretName }}/ca
+      cert_file: /etc/prometheus/secrets/{{ .SecretName }}/crt
+      key_file: /etc/prometheus/secrets/{{ .SecretName }}/key
+      insecure_skip_verify: false
+  tls_config:
+    ca_file: /etc/prometheus/secrets/{{ .SecretName }}/ca
+    cert_file: /etc/prometheus/secrets/{{ .SecretName }}/crt
+    key_file: /etc/prometheus/secrets/{{ .SecretName }}/key
+    insecure_skip_verify: true
```

Comment:
Is there a specific reason why we need this to true?

Reply:
Otherwise certificate verification fails because certificate only has DNS record and we connect via IP here.

Reply:
thanks

Reply:
Could we actually set it to false if we used the api server as a proxy like https://github.com/giantswarm/prometheus-meta-operator/blob/7533e3f59910d4c6a2d65467495783e0a2f961d1/files/templates/additional-scrape-configs.template.yaml#L3 ?

Reply:
We could but if api goes down we loose monitoring of kubelets I guess. I'd like to keep it like this.